File: westcoast/src/main/java/frc/commands/motors/SetMotor.java
package frc.commands.motors;

import edu.wpi.first.wpilibj.SpeedController;
import edu.wpi.first.wpilibj2.command.CommandBase;

public class SetMotor extends CommandBase {
  private SpeedController motor;
  private double value;

  public SetMotor(
    SpeedController motor,
    double value
  ) {
    this.motor = motor;
    this.value = value;
  }

  @Override
  public void initialize() {
  }

  @Override
  public void execute() {
    motor.set(value);
  }

  @Override
  public void end(boolean interrupted) {
    motor.stopMotor();
  }

  @Override
  public boolean isFinished() {
    return false;
  }
}

File: westcoast/src/main/java/frc/commands/motors/SetMotor.kt
package frc.commands.motors

import edu.wpi.first.wpilibj.SpeedController
import edu.wpi.first.wpilibj2.command.CommandBase

class SetMotor(
    private val motor: SpeedController,
    private val value: Double
) : CommandBase() {
  override fun initialize() {}
  override fun execute() {
    motor.set(value)
  }

  override fun end(interrupted: Boolean) {
    motor.stopMotor()
  }

  override fun isFinished(): Boolean {
    return false
  }
}